### PR TITLE
add hash length macro

### PR DESCRIPTION
**Pull request by dmolik**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -1,6 +1,6 @@
 AC_PREREQ(2.63)
 AC_INIT([libvigor], [m4_esyscmd([./version.sh])], [bugs@niftylogic.com])
-AX_ABI_VERSION([2], [0], [1])
+AX_ABI_VERSION([2], [1], [1])
```
Comment (jhunt) [inline configure.ac]: I'm not sure this version bump is correct.  Given that the structures are changing, might we run into a problem where compiled binaries are referencing 2.x.x ABI, but passing a hash_t structure (statically allocated in the image) that is 8-32 bytes short?

**Comment by jhunt** [comment]
Since this is going to necessitate an ABI change, should we also knock out #10?

**Comment by dmolik** [comment]
both are good points. I'll pivot and adjust.

File: configure.ac [hunk 2]
```diff
@@ -1,6 +1,6 @@
 AC_PREREQ(2.63)
 AC_INIT([libvigor], [m4_esyscmd([./version.sh])], [bugs@niftylogic.com])
-AX_ABI_VERSION([2], [0], [1])
+AX_ABI_VERSION([2], [0], [0])
```
Comment (jhunt) [inline configure.ac:3]: Shouldn't this be 3.0.0 now?

Comment (dmolik) [inline configure.ac:3]: it's first minus last for and the last number becomes the middle lib/libvigor.1.1.0, I still don't have a proper mnemonic device to explain it. 

the above change produces
we're going to changing the abi to lib/libvigor.2.0.0 and libvigor.2

**Pull request merged**